sulong/projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/nodes/asm/syscall/LLVMSyscallNode.java

```diff
@@ -41,29 +41,29 @@
 import com.oracle.truffle.llvm.runtime.memory.LLVMSyscallOperationNode;
 import com.oracle.truffle.llvm.runtime.nodes.api.LLVMExpressionNode;
 
-@NodeChild("rax")
-@NodeChild("rdi")
-@NodeChild("rsi")
-@NodeChild("rdx")
-@NodeChild("r10")
-@NodeChild("r8")
-@NodeChild("r9")
-public abstract class LLVMAMD64SyscallNode extends LLVMExpressionNode {
+@NodeChild("syscallNum")
+@NodeChild("arg1")
+@NodeChild("arg2")
+@NodeChild("arg3")
+@NodeChild("arg4")
+@NodeChild("arg5")
+@NodeChild("arg6")
+public abstract class LLVMSyscallNode extends LLVMExpressionNode {
     protected static final int NUM_SYSCALLS = 332;
 
-    protected LLVMSyscallOperationNode createNode(long rax) {
-        return LLVMLanguage.getLanguage().getCapability(PlatformCapability.class).createSyscallNode(rax);
+    protected LLVMSyscallOperationNode createNode(long syscallNum) {
+        return LLVMLanguage.getLanguage().getCapability(PlatformCapability.class).createSyscallNode(syscallNum);
     }
 
-    @Specialization(guards = "rax == cachedRax", limit = "NUM_SYSCALLS")
-    protected long cachedSyscall(@SuppressWarnings("unused") long rax, Object rdi, Object rsi, Object rdx, Object r10, Object r8, Object r9,
-                    @Cached("rax") @SuppressWarnings("unused") long cachedRax,
-                    @Cached("createNode(rax)") LLVMSyscallOperationNode node,
+    @Specialization(guards = "syscallNum == cachedsyscallNum", limit = "NUM_SYSCALLS")
+    protected long cachedSyscall(@SuppressWarnings("unused") long syscallNum, Object arg1, Object arg2, Object arg3, Object arg4, Object arg5, Object arg6,
+                    @Cached("syscallNum") @SuppressWarnings("unused") long cachedsyscallNum,
```

Comment:
I'd rename `cachedsyscallNum` to `cachedSyscallNum`.

Reply:
Done.

Reply:
(along with the travis formatting fixes)